### PR TITLE
Dependency cleanup

### PR DESCRIPTION
**Pull request by ziw-liu**
Remove unnecessary dependencies.

### CODE REVIEW

**Comment by ziw-liu** [comment]
Note: `imageio` block was removed because:
1. napari 0.4.17 is released with the blocking patch
2. `imageio` has a new release

File: setup.cfg [hunk 1]
```diff
@@ -40,17 +40,11 @@ install_requires =
 	waveorder==1.0.0rc0
 	pycromanager==0.19.2
 	click>=8.0.1
-	pyyaml>=5.4.1
```
Comment (talonchandler) [inline setup.cfg:43]: Do you need this for the gui dump?

File: setup.cfg [hunk 2]
```diff
@@ -40,17 +40,11 @@ install_requires =
...
-	tqdm>=4.61.1
```
Comment (talonchandler) [inline setup.cfg:44]: The zarr converter uses this.

Comment (talonchandler) [reply setup.cfg:44]: And it's pretty light, so I don't mind keeping it

File: setup.cfg [hunk 3]
```diff
@@ -40,17 +40,11 @@ install_requires =
...
-	opencv-python>=4.5.3.56 # candidate for deletion 1.0
```
Comment (talonchandler) [inline setup.cfg:45]: Just confirmed that it seems to be only a wo dependency. LGTM

File: setup.cfg [hunk 4]
```diff
@@ -40,17 +40,11 @@ install_requires =
...
 	natsort>=7.1.1
 	colorspacious>=1.1.2
 	pyqtgraph>=0.12.3
-	superqt>=0.2.4
```
Comment (talonchandler) [inline setup.cfg:49]: The visuals sliders (that we've hidden for now) use this. Fine to remove this dep for now.

Comment (talonchandler) [reply setup.cfg:49]: You'll just need to make sure to remove the import from main_widget.

File: setup.cfg [hunk 5]
```diff
@@ -40,17 +40,11 @@ install_requires =
...
 	napari-ome-zarr>=0.3.2 # drag and drop convenience
-	qtpy # napari best practice (PyQt5 and PySide2 wrapper)
```
Comment (talonchandler) [inline setup.cfg:51]: We need this, right?

File: setup.cfg [hunk 6]
```diff
@@ -40,17 +40,11 @@ install_requires =
...
 	napari[all] # uses pip-installed PyQt5 by default
-	imageio!=2.11.0,!=2.22.1 # matching napari PR #5168, candidate for removal after the upstream issue is fixed
```
Comment (talonchandler) [inline setup.cfg:53]: Can we delete it entirely? Or just remove the specific versions?

File: setup.cfg (hunk 2 above)
Comment (ziw-liu) [reply setup.cfg:44]: napari already depends on this and we depend on napari, e.g. we do not depend on NumPy for the same reason

Comment (talonchandler) [reply setup.cfg:44]: Gotcha! Thanks.

**Comment by ziw-liu** [comment]
@talonchandler Please refer to the description of https://github.com/mehta-lab/recOrder/pull/283/commits/347e6fb5c691514c138f4d1fe2b4f4bb6bc85f4b and napari's [requirements](https://github.com/napari/napari/blob/ceb0a0303490c78d57d537861e2a8618526d2d75/setup.cfg#L44-L80) for your concerns.

File: setup.cfg (hunk 6 above)
Comment (ziw-liu) [reply setup.cfg:53]: See: https://github.com/mehta-lab/recOrder/pull/283#issuecomment-1319358403

**Comment by talonchandler** [comment]
> @talonchandler Please refer to the description of [347e6fb](https://github.com/mehta-lab/recOrder/commit/347e6fb5c691514c138f4d1fe2b4f4bb6bc85f4b) and napari's [requirements](https://github.com/napari/napari/blob/ceb0a0303490c78d57d537861e2a8618526d2d75/setup.cfg#L44-L80) for your concerns.

Thanks for the links and for looking into this!

**Review by talonchandler — APPROVED** [review]
Just tested a from-scratch installation on hummingbird. Works great!

**Pull request merged**